### ISSUE TITLE
Template inheritance?

### ISSUE THREAD

**Issue by joefiorini** [Shopify/liquid#146]
Django templates and Swig both allow template inheritance. This allows you to define blocks of content in one template (say a layout file) and then extend that template in another where you can place the content that goes into these blocks, very similar to `yield` and `content_for` in Rails. For example:
#### Layout

``` liquid
...
<title>{% block title %}Default Title{% endblock %}</title>
...
```
#### Template

``` liquid
{% extend layout %}

{% block title %}Real Title{% endblock %}
```

This would allow us to specify layouts for our templates which would be much easier than having to pass the layout into render. Is there any interest in having this feature in Liquid?

**Comment by jhabdas** [comment]
It's been a few years since this was closed. @joefiorini are you still using Cadenza? If so, what has your experience been like?

FWIW Hugo introduced template inheritance in June with [`0.16`](https://github.com/spf13/hugo/releases/tag/v0.16). I gave it a whirl by creating [a Hugo theme](https://github.com/comfusion/after-dark) using template inheritance and it felt pretty good compared to everything using partials.
